service/operator/service.go

```diff
@@ -123,6 +126,11 @@ func (s *Service) addFunc(obj interface{}) {
 		s.Logger.Log("error", "could not convert to azuretpr.CustomObject")
 	}
 
+	_, err := azure.NewClients(s.AzureConfig)
+	if err != nil {
+		s.Logger.Log("error", "could not create azure api clients '%#v'")
+	}
```

Comment:
This is what I am saying. This is business logic. "renewing tokens" is a primitive a service implementation should provide. A clients job is to talk to some API. Not more.